pkg/artifactory/repositories.go

```diff
@@ -1132,11 +1241,231 @@ func mkResourceSchema(skeema map[string]*schema.Schema, packer PackFunc, unpack
+
+	localDefaultRepoLayoutMap := map[string]string{
+		"alpine":    "simple-default",
+		"bower":     "bower-default",
+		"cran":      "simple-default",
+		"cargo":     "simple-default",
+		"chef":      "simple-default",
+		"cocoapods": "simple-default",
+		"composer":  "composer-default",
+		"conan":     "conan-default",
+		"conda":     "simple-default",
+		"debian":    "simple-default",
+		"docker":    "simple-default",
+		"gems":      "simple-default",
+		"generic":   "simple-default",
+		"gitlfs":    "simple-default",
+		"go":        "go-default",
+		"gradle":    "maven-2-default",
+		"helm":      "simple-default",
+		"ivy":       "ivy-default",
+		"maven":     "maven-2-default",
+		"npm":       "npm-default",
+		"nuget":     "nuget-default",
+		"opkg":      "simple-default",
+		"pub":       "simple-default",
+		"puppet":    "puppet-default",
+		"pypi":      "simple-default",
+		"sbt":       "sbt-default",
+		"vagrant":   "simple-default",
+		"rpm":       "simple-default",
+	}
+
+	if selectRandom {
+		repoLayout = selectRandomFromMapOfStrings(localDefaultRepoLayoutMap)
+	} else {
+		if value, ok := localDefaultRepoLayoutMap[repoType]; ok {
+			repoLayout = value
+		}
+	}
+	return repoLayout
+}
+
+//Return the default repo layout of Remote Repository type
+func getDefaultRemoteRepoLayoutRef(repoType string, opts ...bool) string {
+
+	selectRandom := isSelectRandom(opts...)
+	const defaultRepoLayout = "simple-default"
+	repoLayout := defaultRepoLayout
+
+	remoteDefaultRepoLayoutMap := map[string]string{
+		"alpine":    "simple-default",
+		"bower":     "bower-default",
+		"cran":      "simple-default",
+		"cargo":     "simple-default",
+		"chef":      "simple-default",
+		"cocoapods": "simple-default",
+		"composer":  "composer-default",
+		"conan":     "conan-default",
+		"conda":     "simple-default",
+		"debian":    "simple-default",
+		"docker":    "simple-default",
+		"gems":      "simple-default",
+		"generic":   "simple-default",
+		"gitlfs":    "simple-default",
+		"go":        "go-default",
+		"gradle":    "maven-2-default",
+		"helm":      "simple-default",
+		"ivy":       "ivy-default",
+		"maven":     "maven-2-default",
+		"npm":       "npm-default",
+		"nuget":     "nuget-default",
+		"opkg":      "simple-default",
+		"p2":        "simple-default",
+		"pub":       "simple-default",
+		"puppet":    "puppet-default",
+		"pypi":      "simple-default",
+		"sbt":       "sbt-default",
+		//"vagrant":   "simple-default", //currently not available
+		"vcs": "simple-default", //Type is only available in Remote Repository
+		"rpm": "simple-default",
+	}
+
+	if selectRandom {
+		repoLayout = selectRandomFromMapOfStrings(remoteDefaultRepoLayoutMap)
+	} else {
+		if value, ok := remoteDefaultRepoLayoutMap[repoType]; ok {
+			repoLayout = value
+		}
+	}
+	return repoLayout
+}
+
+//Return the default repo layout of Virtual Repository type
+func getDefaultVirtualRepoLayoutRef(repoType string, opts ...bool) string {
```

Comment:
Random in production code?

Reply:
`opts ...bool` has been introduced to handle optional parameter with default values. This has been introduced only to run test cases with random legitimate data of repo layout. The function has been used for functional & test code. It will return random value only if you pass true (specifically designed to handle test data).

```
func getDefaultLocalRepoLayoutRef(repoType string, opts ...bool) string
func getDefaultRemoteRepoLayoutRef(repoType string, opts ...bool) string
func getDefaultVirtualRepoLayoutRef(repoType string, opts ...bool) string
func getDefaultFederatedRepoLayoutRef(repoType string, opts ...bool) string
```

Reply:
No test code in production please.